tests/test_catalog.py

```diff
@@ -1218,6 +1218,26 @@ def test_get_all_collections(self) -> None:
         self.assertGreater(len(all_collections), 0)
         self.assertTrue(all(isinstance(c, pystac.Collection) for c in all_collections))
 
+    def test_get_single_links_media_type(self) -> None:
+        catalog = TestCases.test_case_9()
+
+        self.assertEqual(
+            catalog.get_single_link("search").href, "./search.html"  # type: ignore
+        )
+        self.assertEqual(
+            catalog.get_single_link(
+                "search", media_type="application/geo+json"
+            ).href,  # type: ignore
+            "./search.json",
+        )
```

Comment:
```suggestion
        html_link = catalog.get_single_link("search")
        assert html_link is not None
        self.assertEqual(html_link.href, "./search.html")

        geojson_link = catalog.get_single_link(
            "search", media_type="application/geo+json"
        )
        assert geojson_link is not None
        self.assertEqual(
            geojson_link.href,
            "./search.json",
        )
```

These type errors are coming up because there is a possibility that `catalog.get_single_link` could return `None`. Rather than ignoring the errors, we've typically been using `assert` statements to verify that the result is not `None.

Reply:
Thank you :) I missed that the assert would fix the type validation. I completely agree with you change suggestions..